pkg/apis/kubeone/helpers.go

```diff
@@ -294,77 +294,92 @@ func (p CloudProviderSpec) CloudProviderInTree() bool {
 // CSIMigrationSupported returns if CSI migration is supported for the specified provider.
 // NB: The CSI migration can be supported only if KubeOne supports CSI plugin and driver
 // for the provider
-func (p CloudProviderSpec) CSIMigrationSupported() bool {
-	return p.External && (p.Azure != nil || p.Openstack != nil || p.Vsphere != nil)
-}
+func (c KubeOneCluster) CSIMigrationSupported() bool {
+	if !c.CloudProvider.External {
+		return false
+	}
 
-// CSIMigrationFeatureGates returns CSI migration feature gates in form of a map
-// (to be used with Kubelet config) and string (to be used with kube-apiserver
-// and kube-controller-manager)
-// NB: We're intentionally not enabling CSIMigration feature gate because it's
-// enabled by default since Kubernetes 1.18
-// (https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)
-// This is a KubeOneCluster function because feature gates are Kubernetes-version dependent.
-func (c KubeOneCluster) CSIMigrationFeatureGates(complete bool) (map[string]bool, string, error) {
-	var featureGates map[string]bool
+	_, err := c.csiMigrationFeatureGates(false)
 
-	switch {
-	case c.CloudProvider.Azure != nil:
-		featureGates = map[string]bool{
-			"CSIMigrationAzureDisk": true,
-			"CSIMigrationAzureFile": true,
-		}
-	case c.CloudProvider.Openstack != nil:
-		featureGates = map[string]bool{
-			"CSIMigrationOpenStack": true,
-			"ExpandCSIVolumes":      true,
-		}
-	case c.CloudProvider.Vsphere != nil:
-		featureGates = map[string]bool{
-			"CSIMigrationvSphere": true,
-		}
-	default:
-		return nil, "", fail.ConfigValidation(fmt.Errorf("csi migration is not supported for selected provider"))
-	}
+	return err == nil
+}
 
-	if complete {
-		for _, u := range c.InTreePluginUnregisterFeatureGate() {
-			featureGates[u] = true
-		}
+func (c KubeOneCluster) csiMigrationFeatureGates(complete bool) (map[string]bool, error) {
+	lessThan21Constraint, err := semver.NewConstraint("< 1.21.0")
+	if err != nil {
+		return nil, err
 	}
```

Comment:
Nit (non-blocking): KubeOne 1.5 doesn't support Kubernetes 1.20 so we can drop this.

Reply:
Done, removed that stuff